### DISCUSSION TITLE
Change temporal route path?

### DISCUSSION THREAD

**Discussion by SergioGracia** [remotion-dev/remotion#891] [Q&A] [answered]
Hello everyone!

I would like to know if there is any way to change the default directory where Remotion stores temporary images when rendering a video. Right now everything is dumped to "/tmp" and my idea is to be able to do it to "/data/tmp". I've looked at the documentation but haven't seen anything about it.

Is it possible to change it?

**Comment by JonnyBurger** [comment]
Hi @SergioGracia! Indeed, that is currently hardcoded and cannot be changed.

We can consider this feature - what is your motivation to use a different directory?
Understanding the reason behind this request would surely help us better think through the implementation.

**Reply by SergioGracia** [reply]
Thanks for the reply! To give you some context, I have a unit mounted in /data/tmp with 50 gigabytes free and I would like to use it instead of /tmp since it can become saturated if I launch several video generation processes in parallel.

In fact in some stress tests I've done it reached the point of "cannot write to disk: no space available" relatively quickly.

**Reply by kave-me** [reply]
I have a ~60 gigabyte SSD that has three partitions, something like this:

- `/` (root): 45% of total
- `/home`: 45% of total
- `swap`: 10% of total

 
And a 1 Terabyte hard disc mainly to store almost all of my data there.
 
If I want to make a 20 Gigabytes of video using Remotion I can't have the project on my SSD.
I think for a video of size 20 Gigabyte we would have almost 20 Gigabytes of temporary cache so I should not exceed more than `available_space/2` of my partion space.

If I could have a flag on the CLI or I would be able to set my setting in a `.env`  file to use something like  `/path-to-hd/remotion_tmp` as my tmp directory it would help.

I have a directory in my SSD, mainly I do work on my projects that are in that directory, if I can manage to have Remotion projects in that directory it would be a plus for me too.

**Comment by JonnyBurger** [comment, marked as answer]
@SergioGracia @kave-me I found a solution!

Node.JS will respect the TMPDIR environment variable on macOS and Linux, and the TEMP environment variable on Windows.

So set the `TMPDIR` or `TEMP` environment variable either globally (put it in your .bashrc for example),
or just for the Remotion script (`TMPDIR=/custom/tmp npx remotion ...`)

**Reply by SergioGracia** [reply]
Thankyou!!!! I'll try this as soon as possible. :)
